### PR TITLE
BUGFIX: Refactor Node::getChildNodes rector refactoring 

### PR DESCRIPTION
**Pull request by dlubitz**
Due to the latest changes to FindChildNodesFilter

### CODE REVIEW

File: tests/Rules/NodeGetChildNodesRector/Fixture/all-with-pagination.php.inc [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+use Neos\ContentRepository\Core\Projection\ContentGraph\Node;
+
+class SomeClass
+{
+    public function run(Node $node)
+    {
+        $subgraph = $this->contentRepositoryRegistry->subgraphForNode($node);
+        // TODO 9.0 migration: Try to remove the iterator_to_array($nodes) call.
+
+        foreach (iterator_to_array($subgraph->findChildNodes($node->nodeAggregateId, \Neos\ContentRepository\Core\Projection\ContentGraph\Filter\FindChildNodesFilter::create(nodeTypeConstraints: null)->with(pagination: ['limit' => 10, 'offset' => 100]))) as $node) {
```
Comment (skurfuerst) [inline tests/Rules/NodeGetChildNodesRector/Fixture/all-with-pagination.php.inc:27]:
```suggestion
        foreach (iterator_to_array($subgraph->findChildNodes($node->nodeAggregateId, \Neos\ContentRepository\Core\Projection\ContentGraph\Filter\FindChildNodesFilter::create(pagination: ['limit' => 10, 'offset' => 100]))) as $node) {
```

is what I'd expect here :)

File: tests/Rules/NodeGetChildNodesRector/Fixture/nodetype-filter-with-pagination.php.inc [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+use Neos\ContentRepository\Core\Projection\ContentGraph\Node;
+
+class SomeClass
+{
+    public function run(Node $node)
+    {
+        $subgraph = $this->contentRepositoryRegistry->subgraphForNode($node);
+        // TODO 9.0 migration: Try to remove the iterator_to_array($nodes) call.
+
+        foreach (iterator_to_array($subgraph->findChildNodes($node->nodeAggregateId, \Neos\ContentRepository\Core\Projection\ContentGraph\Filter\FindChildNodesFilter::create(nodeTypeConstraints: 'Neos.Neos:Document')->with(pagination: ['limit' => 10, 'offset' => 100]))) as $node) {
```
Comment (skurfuerst) [inline tests/Rules/NodeGetChildNodesRector/Fixture/nodetype-filter-with-pagination.php.inc:27]:
```suggestion
        foreach (iterator_to_array($subgraph->findChildNodes($node->nodeAggregateId, \Neos\ContentRepository\Core\Projection\ContentGraph\Filter\FindChildNodesFilter::create(nodeTypeConstraints: 'Neos.Neos:Document', pagination: ['limit' => 10, 'offset' => 100]))) as $node) {
```

is what I'd expect here

**Review by skurfuerst — COMMENTED** [review]
@dlubitz nitpicks ;) All the best <3

File: tests/Rules/NodeGetChildNodesRector/Fixture/nodetype-filter-with-pagination.php.inc (hunk 2 above)
Comment (dlubitz) [reply tests/Rules/NodeGetChildNodesRector/Fixture/nodetype-filter-with-pagination.php.inc:27]: 😮

Comment (dlubitz) [reply tests/Rules/NodeGetChildNodesRector/Fixture/nodetype-filter-with-pagination.php.inc]: Done

**Review by skurfuerst — APPROVED** [review]

**Pull request merged**